### ISSUE TITLE
[Feature Request] Missing "items-length" property in v-data-iterator component

### ISSUE THREAD

**Issue by lucasmcht** [vuetifyjs/vuetify#19486]
### Problem to solve
Currently, pagination has to be done locally, while it may be required in some use cases to paginate using API. Making this property available would greatly simplify this.

### Proposed solution
Add 'items-length' property, with the same as in the 'v-data-table-server' component.

**Comment by Maqsyo** [comment]
I would really appreciate this feature

**Comment by lucasmcht** [comment]
The fact that this old functionality is no longer present is really handicapping for us, if we propose a PR to add an "itemsLength" prop, will it only be read or is it not in the objectives of Vuetify to add this functionality?

**Comment by kor3k** [comment]
same problem applies to v-data-table

https://github.com/vuetifyjs/vuetify/discussions/20035#discussioncomment-10522051

**Comment by johnleider** [comment]
Resurrecting this issue. Can someone please show me a playground of how you would use this? Is it as simple as just adding a new prop and forcing itemsLength computed property to be either it or flatItems?

https://github.com/vuetifyjs/vuetify/blob/9d30fa56be862ed0d78a2145b0375a4609586fa9/packages/vuetify/src/components/VDataIterator/VDataIterator.tsx#L124

**Comment by lucasmcht** [comment]
@johnleider Yes that's the idea, currently we apply the following patch (using [patch-package](https://www.npmjs.com/package/patch-package)) to make it work :

```diff --git a/node_modules/vuetify/lib/components/VDataIterator/VDataIterator.mjs b/node_modules/vuetify/lib/components/VDataIterator/VDataIterator.mjs
index bdf0dab..54cfd2c 100644
--- a/node_modules/vuetify/lib/components/VDataIterator/VDataIterator.mjs
+++ b/node_modules/vuetify/lib/components/VDataIterator/VDataIterator.mjs
@@ -88,7 +88,7 @@ export const VDataIterator = genericComponent()({
     const {
       flatItems
     } = useGroupedItems(sortedItems, groupBy, opened);
-    const itemsLength = computed(() => flatItems.value.length);
+    const itemsLength = computed(() => props.itemsLength ?? flatItems.value.length);
     const {
       startIndex,
       stopIndex,
@@ -102,14 +102,16 @@ export const VDataIterator = genericComponent()({
       itemsPerPage,
       itemsLength
     });
-    const {
-      paginatedItems
-    } = usePaginatedItems({
-      items: flatItems,
-      startIndex,
-      stopIndex,
-      itemsPerPage
-    });
+    let paginatedItems;
+    if (!!props.itemsLength) paginatedItems = flatItems;
+    else {
+      paginatedItems = usePaginatedItems({
+        items: flatItems,
+        startIndex,
+        stopIndex,
+        itemsPerPage
+      }).paginatedItems;
+    }
     const paginatedItemsWithoutGroups = computed(() => extractRows(paginatedItems.value));
     const {
       isSelected,
diff --git a/node_modules/vuetify/lib/components/VDataIterator/composables/items.mjs b/node_modules/vuetify/lib/components/VDataIterator/composables/items.mjs
index 3bfeaf7..28ca1bb 100644
--- a/node_modules/vuetify/lib/components/VDataIterator/composables/items.mjs
+++ b/node_modules/vuetify/lib/components/VDataIterator/composables/items.mjs
@@ -15,6 +15,10 @@ export const makeDataIteratorItemsProps = propsFactory({
     type: [String, Array, Function],
     default: null
   },
+  itemsLength: {
+    type: Number,
+    default: null,
+  },
   returnObject: Boolean
 }, 'DataIterator-items');
 export function transformItem(props, item) {

```

**Comment by adamskeeled** [comment]
@johnleider This seems to be stagnant, any idea when this could go forward? Ran into this issue yesterday

**Comment by J-Sek** [comment]
What's the use case? Lazy loading collection? Fully server-side pagination & filtering? Without understanding the use case I am inclined to classify this as a misunderstanding.
- pagination can be easily instructed to have any `length`
- alternatively, you can populate `items` with dummy items (placeholders) to be swapped once they get loaded
...either way, I don't see how faking collection size inside the component helps achieve anything. I would assume you would need much more than this in a realistic scenario.

**Comment by lucasmcht** [comment]
Hello @J-Sek, 

Our use case is quite simple, we have a **paginated REST API endpoint**, and we are using it to display a certain amount of items on the page. The endpoint returns the total amount of items, so that the client can compute the next request to do, and we want to pass this "total" to the `v-data-iterator`, so that it can use it for it's internal pagination. The `v-data-iterator` component already have internal pagination, we just want to be able to control the way that the total amount of items is computed. This feature was already included back in vuetify **2.x.x**. 

Our final use case is server side pagination, although I think that both the use cases that you mentionned are important to have for this component either way.

**Comment by kor3k** [comment]
same use case as @lucasmcht . 

wanted to list & paginate server data with `v-data-iterator`. it was not possible, so i tried to use `v-data-table(-server)` and override bottom slot. it was not possible too because none of these [exposes slot props](https://github.com/vuetifyjs/vuetify/discussions/20035#discussioncomment-10522051) `itemsLength` there.

**Comment by J-Sek** [comment]
Could you remix this [demo](https://play.vuetifyjs.com/#eNqVVm1v40QQ/iujAKoDtpM0zRVMc+gEHJQTUj/wBbWVbmuPk6X22tpdJ81F+e/Mrl/itXucqKrE3nnmmffJ3h8nSsazd2UZ7iqcRJMbjXmZMY1vHwTAzS5gZQl6izmuHyYJk88PEyuxsrgQmnGBEnL2Eux5oreEup7PO5CFJUyzgGuUTBeyPQeI6ChXQYkyKNnG8NMXF0zzQoRWdofyjiQPk6ESYRXKHcpb8+bIs4IlXGwI0Tz1pUY52LGsMtYEy3vUncPkcpsD+MpwoOzJSJrwXf8dQOlDZghzLoIt8s1WR3C5WpUvfdPmL86YMr4nQZrhC7CMb0QQo6DUwD+V0jw9NK+upmPfprSUxUaiUkHMZVxlTAIXCZIi+WAcb2uxfJiA4p+Md4sVlQVmbiwzCqYf+cwp/ygdCaasyjSxHW0uFZx6pW6z818JGYZi4IO0mI9gL1k5yAlsWLBskqY0k3rEVjclk8nwGGAXpIUkG55x2wc+pYTVIQyLZLvoGQ8E5q/J2uBsiiO4mn/z47n9I1jO5+PCW07NtVU0VkPJ9qHbgD2kqp5a8Mevjx0+ZlkhOaoTtE8fx+pugccl/v9FT4uCsu9SUAd2s+ra2wV5kWDmTrOZ76GrUYZiY5v0T6a3YYw883ShWWZnGmbwmW0wdYn68b4Syc1ssH/a7UXn3fqqdx2d0LKjZ4eGXlUsealBoa6MmNZIXhZSwxH2TMdbHySyWPMdmqcUTpDKIocLWqgXRp/mvhBKQ0LzilIrWMN97cQRTAtEcPFeFp9QwN/FppL6wu/qG8Fi9QOc/AH8D8yyAzwhEw52eb0aYz9w/YG5nKvF92PcrzSFXDq4yzeXY9xvtFJRPlHIiWt89WYMvo0RYoLmoJhI9jzeugaW12Odn6syZs/oks9fiez3QuAhLvInB3o1fyW4XwpRuTm4WrWxPZ4r9J7svru7pQIdawKmDiKGFKnK4B1NR1KN+80Ip2mLBSq+rqQAgXu4ow7gCj3a0UW2Q1i/PcNohaD+i+dYVNrzpgNZ64vdcOSJZ4xCAIspfOuYHqugSEihVvzuC9h6e6+7pgxVxmP0pnXDuthmENGwW70GbC35tdn1GoIF/AQV/QylNFMJREZAfGe2JhmUSMsSnYl9sJMfNfT1ZqDcnrVPPqzm8+6gFdUlPPXHrHcxIH9pIL37R4uuxc2toBFpWdFC6YS9/VPLjcWz+LyRrLgeeq8pnqkTzWvTev3s08RZLxsuuzRMWRsuH+hGYrrAuGaNe3Qw9al9eZ5jwilDERhPO4q0EmSb3OhUvtieTdyhvf2Q+4avljRdH9o2/xxNSHdA4bWla+pl6Pvd20t9Z6f+jW0A5/ye/TBHrXzoZMoy1XhZV5wqfTOr9zFt4snJnyzDVXi5mJiHxZz+J75V8p17bX32+C8UYVMd) to visualize the problem?

**Comment by adamskeeled** [comment]
@J-Sek In my case, I have a fully server-side pagination with filtering. I have multiple views (like table view and card view) to display my data, and for consistency sake, I want the same pagination UI on both. Since the table view(with VDataTableServer) already does everything automatically with the VDataTableFooter, I wanted to also insert a VDataTableFooter in the footer slot of VDataIterator(for the card view). However, since VDataIterator doesn't have a `items-length` property and the footer slot doesn't expose it, I can't pass that along to the VDataTableFooter.

The only workaround to this would be to manually recreate the UI and logic behind the VDataTableFooter with a VSelect and VPagination. But of course I'd prefer to just use the VDataTableFooter directly.

If it makes more sense, there could even be a VDataIteratorServer (just like how VDataTable does it for VDataTableServer)
